### PR TITLE
Docs

### LINKED ISSUE #49: Add support for collections

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

- [x] PHP attributes
- [x] Type casting
- [x] Attribute reslover factory
- [x] Object factory
- [x] Mapping

### CODE REVIEW

File: docs/guide/en/creating-own-attributes.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        }
+
+        $value = rand($attribute->getMin(), $attribute->getMax());
+
+        return Result::success($value);
+    }
+}
+```
+
+## Usage one class for attribute and resolver
```
Comment (samdark) [inline docs/guide/en/creating-own-attributes.md:141]:
```suggestion
## Using a single class for both attribute and resolver
```

File: docs/guide/en/creating-own-attributes.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+If your attribute simple and don't required dependencies, you can combine attribute and his resolver in one class.
```
Comment (samdark) [inline docs/guide/en/creating-own-attributes.md:143]:
```suggestion
If your attribute is simple and doesn't require dependencies, you can combine attribute and its resolver in a single class.
```

**Review by samdark — APPROVED** [review]

File: docs/guide/en/general-usage.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+$object = $hydrator->create(Car::class, [
+    'name' => 'Ferrari',
+    'engine.name' => 'V8',
+]);
+```
+
+That would pass the `name` constructor argument of the `Car` object and create a new `Engine` object for `engine`
+argument passing `V8` as the `name` argument to its constructor.
+
+// TODO: strict mode
```
Comment (samdark) [inline docs/guide/en/general-usage.md:57]: Not sure how to describe it best.

File: docs/guide/en/mapping.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+Hydrator allows you to map data: 
+
+```php
+use Yiisoft\Hydrator\Hydrator;
+use Yiisoft\Hydrator\ArrayData;
+
+$hydrator = new Hydrator();
+
+$map = ['title' => 'header', 'body' => 'text'],;
```
Comment (vjik) [inline docs/guide/en/mapping.md:41]:
```suggestion
$map = ['title' => 'header', 'body' => 'text'];
```

File: docs/guide/en/typecasting.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+```php
+final class Lock
+{
+    public function __construct(
+        private string $name,
+        private bool $isLocked
+    )
+    {
+    }
```
Comment (vjik) [inline docs/guide/en/typecasting.md:13]:
```suggestion
    public function __construct(
        private string $name,
        private bool $isLocked
    ) {}
```

**Review by samdark — APPROVED** [review]

**Pull request merged**